Cognite.Simulator.Tests/UtilsTests/StateStoreTest.cs

```diff
@@ -0,0 +1,69 @@
+using System.Collections.Generic;
+using System.Text.Json;
+
+using Cognite.Extractor.StateStorage;
+using Cognite.Simulator.Utils;
+
+using Xunit;
+
+namespace Cognite.Simulator.Tests.UtilsTests
+{
+    public class StateStoreTest
+    {
+        [Fact]
+        public void DependencyFile_Serialization()
+        {
+            // Arrange
+            var dependencyFile = new DependencyFile
+            {
+                Id = 123,
+                FilePath = "/some/path/doc.txt",
+                Arguments = new Dictionary<string, string> { { "key", "value" } }
+            };
+
+            var mapper = StateStoreUtils.BuildMapper();
+
+            // Act
+            var raw = StateStoreUtils.BsonToDict(mapper.ToDocument(dependencyFile));
+            var json = JsonSerializer.Serialize(raw);
+
+            // Assert
+            var expectedJson = "{\"_id\":123,\"arguments\":{\"key\":\"value\"},\"file-path\":\"/some/path/doc.txt\"}";
+            Assert.Equal(expectedJson, json);
+        }
+
+        [Fact]
+        public void ModelStateBasePoco_Serialization()
```

Comment:
what value do we gain from this (`ModelStateBasePoco_Serialization`) test?

Reply:
when we add props in the future, it might break and give us a hint.
if we remove, it also breaks, reminding that we might need to think of migration